shoop/campaigns/models/campaigns.py

```diff
@@ -74,22 +66,16 @@ def is_available(self):
                 return False
         return True
 
-    def save(self, **kwargs):
-        if self.discount_percentage and self.discount_amount_value:
-            raise ValidationError(_("You should only define either discount percentage or amount."))
-
-        if not (self.discount_percentage or self.discount_amount_value):
-            raise ValidationError(_("You must define discount percentage or amount."))
-
-        super(Campaign, self).save(**kwargs)
-
 
 class CatalogCampaign(Campaign):
     _queryset = None
 
     admin_url_suffix = "catalog_campaigns"
     conditions = models.ManyToManyField('ContextCondition', blank=True, related_name='campaign')
     filters = models.ManyToManyField('CatalogFilter', blank=True, related_name='campaign')
+    effects = models.ManyToManyField(
+        CampaignEffect, verbose_name=_("effects"), blank=True,
+        help_text=_("When campaign is triggered, these effects give end result."), related_name='catalog_campaigns')
```

Comment:
Effect should probably be linked to campaign through a `campaign` FK in `CampaignEffect` model. See my comments above.

That would mean changing the campaign to PolymorphicModel though, or using generic foreign key.